spikes/encoder.py

```
@@ -303,3 +303,46 @@ def _fir_filter(self):
 
         return flow
+
+class StepForward:
+    """
+    Step Forward Algorithm (SF) encoder.
+
+    Iteratively accumulates absolute values across time steps and emits a spike
+    whenever the accumulated value crosses the threshold, resetting the accumulator
+    by subtracting the threshold. This provides a rate-based encoding scheme.
+    """
+    def __init__(self, data, threshold=0.5):
```

Ruff (0.16.0)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)